modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/sync/PodcastSyncProcess.kt

```diff
@@ -299,28 +299,31 @@ class PodcastSyncProcess(
     private fun syncUpNext() = Completable.create { emitter ->
         val startTime = SystemClock.elapsedRealtime()
         val workRequestId = UpNextSyncWorker.enqueue(syncManager, context)
-        workRequestId?.let {
+        if (workRequestId == null) {
+            logTime("Refresh - sync up next, work request id is null", startTime)
+            emitter.onComplete()
+        } else {
             ProcessLifecycleOwner.get().lifecycleScope.launch {
-                WorkManager.getInstance(context).getWorkInfoByIdFlow(it).first { it.state.isFinished }
+                WorkManager.getInstance(context).getWorkInfoByIdFlow(workRequestId).first { it.state.isFinished }
+                logTime("Refresh - sync up next", startTime)
                 emitter.onComplete()
             }
-        } ?: emitter.onComplete()
-        LogBuffer.i(LogBuffer.TAG_BACKGROUND_TASKS, "Refresh - sync up next - ${String.format(Locale.ENGLISH, "%d ms", SystemClock.elapsedRealtime() - startTime)}")
```

Comment:
Logging the time outside of the lambda meant that it wasn't the time to run the Up Next sync and only the time to run the method.